### PR TITLE
main-7.0.x backport of JA4 

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #11252  

Backport JA4 changes

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [7010](https://redmine.openinfosecfoundation.org/issues/7010)

Describe changes:
- Backport of [6379](https://redmine.openinfosecfoundation.org/issues/6379)

Updates:
- Address CI issues: formatting, improper rst indentation

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1875

### CODE REVIEW

File: src/output-json-quic.c [hunk 1]
```diff
@@ -93,6 +97,14 @@ static OutputInitResult OutputQuicLogInitSub(ConfNode *conf, OutputCtx *parent_c
         SCFree(quiclog_ctx);
         return result;
     }
+
+    /* In 7.0.x, ja4 hash is only logged when requested */
+    quiclog_ctx->log_ja4 = false;
+    ;
```
Comment (victorjulien) [inline src/output-json-quic.c:103]: nit

**Comment by victorjulien** [comment]
needs a rebase for green CI

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 20999

**Comment by jlucovsky** [comment]
Continued in #11276

**Pull request closed without merging**